### PR TITLE
reworked original 2.x version of xrange into range

### LINKED ISSUE #50: change xrange to be range
As per 3.0 xrange is renamed to range and range (list) is gone.
As per 3.0 range must conform to int/long unification and return longs.

Reference: https://docs.python.org/3.4/library/stdtypes.html#ranges

### PR DESCRIPTION
**Pull request by paweljasinski**
Fixes #50.

added applicable test from stdlib
added convenience methods to iptest/assert_util.py to mimic unittest
added slice operator
added == != < <= > >= operators
added `__contains__` method
added count and index methods of collections.abc.Sequence
changed start, stop and step to be attributes
added `__hash__`
added support for objects implementing `__index__`
Converter.TryConvertToIndex does not ignore throwOverflowError parameter

Outstanding: pickle testing - it depends on itertools which depends on missing language features
When reviewing, go for second commit otherwise rename obfuscates changes.

Test results:
```
rejap@WIN-CUE1I6EN9JB ~/github/ironpython3/bin/Debug
$ ./IronPythonTestConsole.exe  IronPythonTest -noh -labels -full -test:IronPythonTest.Cases.IronPythonCases.test_range
Runtime Environment -
    OS Version: Microsoft Windows NT 6.1.7601 Service Pack 1
  .NET Version: 4.0.30319.18444

***** IronPythonTest.DLL
***** IronPythonCases
***** IronPythonTests
***** test_range
>>>   0.00s testing test_attributes_from_stdlib
>>>   0.27s testing test_comparison_from_stdlib
>>>   0.40s testing test_contains_from_stdlib
>>>   0.47s testing test_count_from_stdlib
>>>   0.55s testing test_empty_from_stdlib
>>>   0.56s testing test_index_from_stdlib
>>>   0.71s testing test_invalid_invocation_from_stdlib
>>>   0.94s testing test_issue11845_from_stdlib
>>>   1.02s testing test_odd_bug_from_stdlib
>>>   1.05s testing test_range
>>>   1.09s testing test_range_collections
>>>   1.11s testing test_range_corner_cases
>>>   1.12s testing test_range_coverage
>>>   1.16s testing test_range_equal
>>>   1.19s testing test_range_from_stdlib
>>>   1.23s testing test_repr_from_stdlib
>>>   1.24s testing test_reverse_iteration_from_stdlib
>>>   1.26s testing test_slice_from_stdlib
>>>   1.30s testing test_strided_limits_from_stdlib
>>>   1.31s testing test_types_from_stdlib
>>>   1.45s testing test_user_index_method_from_stdlib

21 tests passed
Tests run: 1, Passed: 1, Errors: 0, Failures: 0, Inconclusive: 0
  Not run: 0, Invalid: 0, Ignored: 0, Skipped: 0
Elapsed time: 00:00:13.2410000
```

### CODE REVIEW

**Comment by slide** [comment]
@paweljasinski Is this something that still needs to be incorporated? Can you take a look at the merge conflicts?

**Comment by paweljasinski** [comment]
@slide there was a trivial line ending mismatch, no real conflict. Rebased and ready to be applied.
The comment about missing itertools is valid and needs to be addressed as other language feature emerge.

**Comment by simplicbe** [comment]
LGTM

File: Src/IronPython/Lib/iptest/assert_util.py [hunk 1]
```diff
@@ -237,8 +237,27 @@ def AssertInOrNot(l, in_list, not_in_list):
...
+def AssertIn(x, l):
+    Assert(x in l, "%s should be in %s" % (x, l))
+
+def AssertNotIn(x, l):
+    Assert(x not in l, "%s should not be in %s" % (x, l))
+
+def AssertIs(x, t):
+    Assert(x == t, "expected type %s, but got %s" % (t, x))
 
+class AssertReisesCtx():
```
Comment (slide) [inline Src/IronPython/Lib/iptest/assert_util.py]: Should this be AssertR**a**isesCtx?

**Comment by paweljasinski** [comment]
and rebased

**Pull request merged**